x-pack/plugins/index_management/server/routes/api/data_streams/register_get_route.ts

```diff
@@ -6,122 +6,188 @@
 
 import { schema, TypeOf } from '@kbn/config-schema';
 
+import { ElasticsearchClient } from 'kibana/server';
 import { deserializeDataStream, deserializeDataStreamList } from '../../../../common/lib';
+import { DataStreamFromEs } from '../../../../common/types';
 import { RouteDependencies } from '../../../types';
 import { addBasePath } from '../index';
 
-const querySchema = schema.object({
-  includeStats: schema.maybe(schema.oneOf([schema.literal('true'), schema.literal('false')])),
-});
+interface PrivilegesFromEs {
+  username: string;
+  has_all_requested: boolean;
+  cluster: Record<string, boolean>;
+  index: Record<string, Record<string, boolean>>;
+  application: Record<string, boolean>;
+}
+
+interface StatsFromEs {
+  data_stream: string;
+  store_size: string;
+  maximum_timestamp: number;
+}
 
-export function registerGetAllRoute({ router, license, lib: { isEsError } }: RouteDependencies) {
+const enhanceDataStreams = ({
+  dataStreams,
+  dataStreamsStats,
+  dataStreamsPrivileges,
+}: {
+  dataStreams: DataStreamFromEs[];
+  dataStreamsStats?: StatsFromEs[];
+  dataStreamsPrivileges?: PrivilegesFromEs;
+}): DataStreamFromEs[] => {
+  return dataStreams.map((dataStream: DataStreamFromEs) => {
+    let enhancedDataStream = { ...dataStream };
+
+    if (dataStreamsStats) {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      const { store_size, maximum_timestamp } = dataStreamsStats.find(
```

Comment:
you are right, it can be `undefined` at the end of the line for TypeScript not to complain. But I was thinking that we rely too much on ES returning the same data streams in both requests, but if a data stream is deleted between the requests then this logic would fail. So I added a default empty object instead.